spec/doubleLinkedList_test.js

```diff
@@ -0,0 +1,156 @@
+import chai, { expect } from 'chai'
+import chaiChange from 'chai-change'
+import DoubleLinkedList from '../src/doubleLinkedList'
+
+chai.use(chaiChange)
+
+describe('Double Linked List', () => {
+  'use strict'
+
+  it('exists', () => {
+    expect(DoubleLinkedList).to.be.a('function')
+  })
+
+  describe('insert()', () => {
+    it('Inserts a node (with the provided data) to the tail of the list', () => {
+      const doubleLinkedList = new DoubleLinkedList()
+
+      expect(() => doubleLinkedList.insert('foo'))
+        .to.alter(() => doubleLinkedList.count, { from: 0, to: 1 })
+      expect(doubleLinkedList.getHeadNode().data)
+        .to.equal('foo')
+      expect(() => doubleLinkedList.insert('boo'))
+        .to.alter(() => doubleLinkedList.count, { from: 1, to: 2 })
+       expect(doubleLinkedList.getTailNode().data)
+        .to.equal('boo')
+    })
+  })
+
+  describe('insertFirst()', () => {
+    it('Inserts a node (with the provided data) to the head of the list', () => {
+      const doubleLinkedList = new DoubleLinkedList()
+
+      expect(() => doubleLinkedList.insertFirst('foo'))
+        .to.alter(() => doubleLinkedList.count, { from: 0, to: 1 })
+      expect(() => doubleLinkedList.insertFirst('boo'))
+        .to.alter(() => doubleLinkedList.count, { from: 1, to: 2 })
+        console.log(doubleLinkedList)
+      expect(doubleLinkedList.getHeadNode().data)
+        .to.equal('boo')
+      expect(doubleLinkedList.getTailNode().data)
+        .to.equal('foo')
+    })
+  })
+  describe('contains()', () => {
+    it('Determines whether or not the list contains the provided data', () => {
+      const doubleLinkedList = new DoubleLinkedList()
+      doubleLinkedList.insertFirst('foo')
+      doubleLinkedList.insertFirst('boo')
+
+      expect(doubleLinkedList.contains('foo'))
+        .to.be.true
+      expect(doubleLinkedList.contains('tap'))
+        .to.equal(false)
+    })
+  })
+
+  describe('find()', () => {
+    it('Determines whether or not the list contains the provided data', () => {
+      const doubleLinkedList = new DoubleLinkedList()
+      doubleLinkedList.insertFirst('foo')
+      doubleLinkedList.insertFirst('boo')
+      doubleLinkedList.insertFirst('shoo')
+      doubleLinkedList.insertFirst('test')
+      console.log(doubleLinkedList)
+
+      expect(doubleLinkedList.find('shoo').data)
+        .to.equal('shoo')
+      expect(doubleLinkedList.find('tap'))
+        .to.equal(-1)
+    })
+  })
+
+  describe('clear()', () => {
+    it('Clears the list of all nodes/data', () => {
+      const doubleLinkedList = new DoubleLinkedList()
+      doubleLinkedList.insertFirst('foo')
+      doubleLinkedList.insertFirst('boo')
+      doubleLinkedList.insert('shoo')
+
+      expect(doubleLinkedList.clear())
```

Comment:
This is not a meaningful test - a function without an explicit `return` statement returns `undefined` in JavaScript.  This test will _always_ pass, even if the `clear()` function is broken.  You have a number of tests like this that should be removed (and perhaps replaced by tests that will fail if the function is broken).